taca/cli.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+import click
+import taca.log
+
+from pkg_resources import iter_entry_points
+
+from taca.utils import config
+
+
+@click.group()
+# Priority for the configuration file is: environment variable > -c option > default
+@click.option('-c', '--config-file',
+			  default=os.path.join(os.environ['HOME'], '.taca/taca.yaml'),
+			  envvar='TACA_CONFIG',
+			  type=click.File('r'),
+			  help='Path to TACA configuration file')
+@click.pass_context
+def cli(ctx, config_file):
+	""" Tool for the Automation of Storage and Analyses """
+
+	ctx.obj = {}
+	ctx.obj['config'] = config.load_yaml_config(config_file)
+	log_file = ctx.obj['config'].get('log', {}).get('log_file', None)
+	if log_file:
+		level = ctx.obj['config'].get('log').get('log_level', 'INFO')
+		taca.log.init_logger_file(log_file, level)
+
+
+#Add subcommands dynamically to the CLI
+for entry_point in iter_entry_points('taca.subcommands'):
```

Comment:
First; I get a kick out finding a way to avoid global variables :smile: 

You are right that it does the exact same thing as the existing code. The end result is identical. ``setup.py`` shouldn't change at all.

As I see it, with a little more work, this class could be made highly reusable across projects. This also seems like a more robust solution, less hack-y but that's really all the reasons I need for introducing the perceived added complexity.